src/telemetry/class-parsely-telemetry.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * Parsely_Telemetry class
+ *
+ * @package Parsely
+ * @since 3.0.0
+ */
+
+declare(strict_types=1);
+
+/**
+ * This class comprises the mechanics of setting up the back end tracking instance(s).
+ * Currently, the only supported back end is Automattic Tracks.
+ * This is intended to wrap the internals such that adding / changing back ends has minimal impact on the event hooks we're interested in.
+ */
+class Parsely_Telemetry {
+	/**
+	 * This is determined by our value passed to the `WP_Widget` constructor.
+	 *
+	 * @see https://github.com/Parsely/wp-parsely/blob/e9f1b8cd1a94743e068681a8106176d23857992d/src/class-parsely-recommended-widget.php#L28
+	 */
+	const RECOMMENDED_WIDGET_BASE_ID = 'parsely_recommended_widget';
+
+	/**
+	 * Holds an instance of the class comprising the Automattic Tracks mechanics.
+	 *
+	 * @var Parsely_A8c_Tracks
+	 */
+	private $tracks;
+
+	/**
+	 * Parsely_Telemetry constructor.
+	 */
+	public function __construct() {
+		$this->tracks = new Parsely_A8c_Tracks();
+		$this->tracks->setup();
+		$this->add_event_tracking();
+	}
+
+	/**
+	 * Hook functions into WordPress actions and / filters for which we want to record events.
+	 *
+	 * @todo Move this to a dedicated file or some other organization scheme.
+	 * @todo Use named functions so they can be (conceivably) unhooked if desired.
+	 *
+	 * @return void
+	 */
+	protected function add_event_tracking(): void {
```

Comment:
It feels like this list of events should be held outside of this class, and registered to the class with a `register_event()` or `add_event()` method.

That would give more flexibility in being able to add extra events for testing, and allow others to add new events to be tracked, and better follows the Single Responsibility Principle.


Simplified: 
```php
// Create new telemetry dispatcher with an instance of the system to use:
$telemetry = new Telemetry( new A8c_Tracks() );

// Now register action and filter hooks and their callbacks.
$telemetry->register_event(
	new Telemetry_Event_Action (
		'my_action_hook',
		'my_callable'
	)
);
$telemetry->register_event(
	new Telemetry_Event_Filter (
		'my_filter_hook',
		'my_callable',
		'my_priority',
		'my_args_count'
	)
);
```

A `$events` property inside Telemetry now contains an array/collection of Telemetry_Event_* data objects.